Xamarin.Forms.Platform.Android/Material/MaterialFrameRenderer.cs

```diff
@@ -180,129 +123,179 @@ protected virtual void OnElementChanged(ElementChangedEventArgs<Frame> e)
 			for (var i = 0; i < children.Count; i++)
 			{
-				var visualElement = children[i] as VisualElement;
-				if (visualElement == null)
-					continue;
-				IVisualElementRenderer renderer = Android.Platform.GetRenderer(visualElement);
-				renderer?.UpdateLayout();
+				if (children[i] is VisualElement visualElement)
+				{
+					var renderer = Platform.GetRenderer(visualElement);
+					renderer?.UpdateLayout();
+				}
 			}
 		}
 
 		public override bool OnTouchEvent(MotionEvent e)
 		{
 			if (_gestureManager.OnTouchEvent(e) || base.OnTouchEvent(e))
-			{
 				return true;
-			}
 
 			return _motionEventHelper.HandleMotionEvent(Parent, e);
 		}
 
-		protected virtual void OnElementPropertyChanged(object sender, PropertyChangedEventArgs e)
+		void UpdateShadow()
 		{
-			ElementPropertyChanged?.Invoke(this, e);
+			if (_disposed)
+				return;
 
-			if (e.PropertyName == Frame.HasShadowProperty.PropertyName)
-				UpdateShadow();
-			else if (e.PropertyName == VisualElement.BackgroundColorProperty.PropertyName)
-				UpdateBackgroundColor();
-			else if (e.PropertyName == Frame.CornerRadiusProperty.PropertyName)
-				UpdateCornerRadius();
-			else if (e.PropertyName == Frame.BorderColorProperty.PropertyName)
-				UpdateBorderColor();
+			// set the default elevation on the first time
+			if (_defaultElevation < 0)
+				_defaultElevation = CardElevation;
+
+			if (Element.HasShadow)
+				CardElevation = _defaultElevation;
+			else
+				CardElevation = 0f;
 		}
 
-		void UpdateBackgroundColor()
+		void UpdateCornerRadius()
 		{
 			if (_disposed)
 				return;
 
-			Color bgColor = Element.BackgroundColor;
+			// set the default radius on the first time
+			if (_defaultCornerRadius < 0)
+				_defaultCornerRadius = Radius;
+
+			var cornerRadius = Element.CornerRadius;
+			if (cornerRadius < 0)
+				Radius = _defaultCornerRadius;
+			else
+				Radius = Context.ToPixels(cornerRadius);
 
-			int[][] States =
-			{
-				new int[0]
-			};
-
-			CardBackgroundColor = new global::Android.Content.Res.ColorStateList
-				(
-					States,
-					new int[] { bgColor.IsDefault ? AColor.White : bgColor.ToAndroid() }
-				);
-			
+			UpdateBorder();
 		}
 
-		void UpdateBorderColor()
+		void UpdateBorder()
 		{
 			if (_disposed)
 				return;
 
-			Color borderColor = Element.BorderColor;
+			// set the default stroke properties on the first time
+			if (_defaultStrokeColor == null)
+				_defaultStrokeColor = StrokeColor;
+			if (_defaultStrokeWidth < 0)
+				_defaultStrokeWidth = StrokeWidth;
+
+			var borderColor = Element.BorderColor;
 			if (borderColor.IsDefault)
 			{
-				StrokeColor = AColor.Transparent;
-				StrokeWidth = 0;
+				StrokeColor = _defaultStrokeColor.Value;
+				StrokeWidth = _defaultStrokeWidth;
 			}
 			else
 			{
 				StrokeColor = borderColor.ToAndroid();
-				StrokeWidth = 25;
-			}
 
+				// TODO: this should be 1, but it is high here to demonstrate the layout issue
+				StrokeWidth = (int)Context.ToPixels(10);
```

Comment:
The Android layout needs to be fixed, and to demonstrate the issue, I have left this a large number.